### PR TITLE
compiler: add build option to enable deprecated generated code

### PR DESCRIPTION
**Pull request by dapengzhang0**
The added option `enable_deprecated` is `true` by default in `java_plugin.cpp`, so the generated code for `TestService.java` (`compiler/build.gradle` not setting this option) has all deprecated interfaces and static bindService method.

`./build.gradle` and `examples/build.gradle` set this option explicitly to `false`, so all the other generated classes do not have deprecated code.

Will set `enable_deprecated` to `false` by default in future PR when we are ready.

### CODE REVIEW

File: compiler/src/java_plugin/cpp/java_generator.cpp [hunk 1]
```diff
@@ -427,110 +427,135 @@ enum CallType {
...
+  p->Print(
+      *vars,
+      "/**\n"
+      " * This will be removed in the next release.\n"
+      " * If your code is migrated to gRPC-java v0.15.0 or higher already,\n"
+      " * the following changes to your code are suggested:\n"
+      " * <ul>\n"
+      " *   <li> replace {@code extends/implements $service_name$}"
+      " with {@code extends $service_name$ImplBase};</li>\n"
+      " *   <li> replace usage of {@code $service_name$} with {@code $service_name$ImplBase};"
```
Comment (ejona86) [inline compiler/src/java_plugin/cpp/java_generator.cpp]: or $service_name$Stub, depending on client or server.

**Comment by dapengzhang0** [comment]
partially resolving #1469

**Comment by ejona86** [comment]
@dapengzhang0, nice. LGTM

**Comment by ejona86** [comment]
@dapengzhang0 LGTM

**Pull request merged**